pandas/tests/indexes/test_interval.py

```diff
@@ -327,171 +335,298 @@ def test_get_item(self):
+        assert index.slice_locs(0, 1) == (3, 2)
+        assert index.slice_locs(0, 2) == (3, 1)
+        assert index.slice_locs(0, 3) == (3, 1)
+        assert index.slice_locs(3, 1) == (1, 2)
+        assert index.slice_locs(3, 4) == (1, 0)
+        assert index.slice_locs(0, 4) == (3, 0)
+
+        # increasing overlapping
+        index = IntervalIndex.from_tuples([(0, 2), (1, 3), (2, 4)])
+        for query in queries:
+            pytest.raises(KeyError, index.slice_locs, query)
+
+        # decreasing overlapping
+        index = IntervalIndex.from_tuples([(2, 4), (1, 3), (0, 2)])
+        for query in queries:
+            pytest.raises(KeyError, index.slice_locs, query)
+
+        # sorted duplicates
+        index = IntervalIndex.from_tuples([(0, 2), (0, 2), (2, 4)])
+        for query in queries:
+            pytest.raises(KeyError, index.slice_locs, query)
+
+        # unsorted duplicates
+        index = IntervalIndex.from_tuples([(0, 2), (2, 4), (0, 2)])
+        for query in queries:
+            pytest.raises(KeyError, index.slice_locs, query)
+
+        # another unsorted duplicates
+        index = IntervalIndex.from_tuples([(0, 2), (0, 2), (2, 4), (1, 3)])
+        for query in queries:
+            pytest.raises(KeyError, index.slice_locs, query)
+
+    @pytest.mark.xfail(reason="new indexing tests for issue 16316")
+    def test_get_indexer_for_interval_updated_behavior(self):
+
+        index = IntervalIndex.from_tuples(
+            [(0, 2.5), (1, 3), (2, 4)], closed='right')
+
+        # single queries
+        queries = [Interval(1, 3, closed='right'),
+                   Interval(1, 3, closed='left'),
+                   Interval(1, 3, closed='both'),
+                   Interval(1, 3, closed='neither'),
+                   Interval(1, 4, closed='right'),
+                   Interval(0, 4, closed='right'),
+                   Interval(1, 2, closed='right')]
+        expected = [1, -1, -1, -1, -1, -1, -1]
+
+        for query, expected_result in zip(queries, expected):
+            result = index.get_indexer([query])
+            expect = np.array([expected_result], dtype='intp')
+            tm.assert_numpy_array_equal(result, expect)
+
+        # multiple queries
+        queries = [
+            [Interval(2, 4, closed='right'), Interval(1, 3, closed='right')],
+            [Interval(1, 3, closed='right'), Interval(0, 2, closed='right')],
+            [Interval(1, 3, closed='right'), Interval(1, 3, closed='left')],
+            index]
+        expected = [[2, 1], [1, -1], [1, -1], [0, 1, 2]]
+
+        for query, expected_result in zip(queries, expected):
+            result = index.get_indexer(query)
+            expect = np.array(expected_result, dtype='intp')
+            tm.assert_numpy_array_equal(result, expect)
+
+    @pytest.mark.xfail(reason="new indexing tests for issue 16316")
+    def test_get_indexer_for_ints_and_floats_updated_behavior(self):
+
```

Comment:
same; anytime you are doing a loop you almost always want to parametrize

Reply:
just a little tidbit on why.

by putting everything in a loop, you MUST debug in order and you have no idea what currently works and what does not. parametrize basically makes little tests out of everything, so it is very easy to see where things work and where they don't. furthermore you can then debug with a very very narrow test and get to exactly what you need w/o going thru lots of steps.